protos/action/action.proto

```diff
@@ -136,12 +136,16 @@ message LandResponse {
     ActionResult action_result = 1;
 }
 
-message RebootRequest {}
+message RebootRequest {
+    uint32 mav_component = 1;
```

Comment:
```suggestion
    uint8 mavlink_component_id = 1; // MAVLink component ID of component to reboot (0 for all).
```

Reply:
Sorry I commented [there](https://github.com/mavlink/MAVSDK/pull/1138#discussion_r451633446). I would not leak the mavlink compid here.

IMO `action.reboot()` talks to the autopilot. Because one may want to reboot everything or only the autopilot, maybe we could have a boolean, like `action.reboot(boolean autopilot_only)`, and by default it would just reboot everything. Then if one needs a way to reboot a camera or a gimbal, I would put that into `drone.camera.reboot()` or `drone.gimbal.reboot()`.

Reply:
Oh, right, I forgot about your suggestion. Ok so we change it to only reboot autopilot, so we don't need to change proto except for a commend saying "only reboots the autopilot"?

Reply:
If I remember correctly, @LorenzMeier added that feature, and the intent was to reboot everything. Hence my suggestion of defaulting to everything, but adding a boolean to limit that to `autopilot_only`. So that we don't break backward compatibility :innocent:. Would that be an acceptable API in your opinion?

Reply:
When 'mavlink_component_id' is 0 or MAV_COMP_ID_ALL, it reboots everything just as it was doing before. So backward compatibility can be intact if we put the default 'mavlink_component_id' as 0.

Reply:
Yes, at the MAVLink level. But the MAVSDK API does not have to expose that there is a "MAVLink component id" at all. We could say that `drone.action.reboot()` sends the reboot command with compid MAV_COMP_ID_ALL, and `drone.action.reboot(true)` (i.e. `autopilot_only = true`) would only reboot the autopilot. So that a user of MAVSDK does not have to know about the concept of component id. Does that make sense?

Reply:
@JonasVautherin ok what should we do?

Reply:
```suggestion
    bool autopilot_only = 1;
```

Because [bools default to 'false'](https://developers.google.com/protocol-buffers/docs/proto3#default) in protobuf, this will not break backward compatibility, but it will allow rebooting only the autopilot (when passing 'true'), which is the new use-case.

That would be my suggestion :innocent:.

Reply:
ok sounds good.